packages/svelte/src/compiler/phases/2-analyze/visitors/ConditionalExpression.js

```diff
@@ -0,0 +1,19 @@
+/** @import { ConditionalExpression } from 'estree' */
+/** @import { Context } from '../types' */
+
+import { mark_subtree_dynamic } from './shared/fragment.js';
+
+/**
+ * @param {ConditionalExpression} node
+ * @param {Context} context
+ */
+export function ConditionalExpression(node, context) {
+	// In legacy mode, we treat conditionals inside the template as not inlinable so patterns
+	// such as BROWSER ? foo : bar, continue to work during hydration
+	if (context.state.expression && !context.state.analysis.runes) {
```

Comment:
AFAIK this worked in runes mode, too, before, so we can't undo this (breaking change)